### DISCUSSION TITLE
Custom start time for matches RFC

### DISCUSSION THREAD

**Discussion by robigan** [evroon/bracket#519] [Ideas]
Any RFCs on adding custom start time functionality for matches @evroon? Could be handled either via a flag that states that a match is using custom scheduling. Also this would be for multi day scheduling, so the planning page would need to show the date too.

**Comment by robigan** [comment]
Also what is up w/ https://github.com/evroon/bracket/blob/master/backend/bracket/schema.py#L100-L103

**Comment by evroon** [comment]
What do you mean by RFC?

**Reply by robigan** [reply]
Request For Comments

**Comment by robigan** [comment]
I was thinking, there's support for custom margin and custom duration. Custom times could make use of the custom margin to update the times

**Comment by evroon** [comment]
I still don't really understand what your goal is. If it is to have matches start later than they would be planned back-to-back, I think it would make more sense to implement a "break" feature. Custom times are going to be annoying to be used, not only because we already have custom duration and custom margins, but also because you would easily forget to update them if something else in the planning changes.

Other systems also implement this custom duration, custom margin and break method of adjusting the planning.

**Reply by robigan** [reply]
Huh, what is this break method though?

**Reply by evroon** [reply]
It's just a break (like a pause) between matches when you want to eat lunch or something.

**Reply by robigan** [reply]
Yeah but how would we implement that? In the form of a match object itself? Or a flag on a match object? I would recommend having a match object that would act as a break with two diff modes. Either allow for the updating of start_time manually of that break match or for the second mode just use custom_duration/duration as the break time, so it becomes relative.

**Reply by robigan** [reply]
Although, having said this, it'd be eaiser to not make a match type of pause, but rather create a whole new object that is part of the rounds/stage_items that gets factored into the time calculations using position_in_schedule.

**Reply by TheKringel** [reply]
Just my two cents on this as newbie on this.

Yes, a break/pause module would be nice to have.
For example I'm planning Icehockey tournaments for kids. We usually play 3 matches with e.g. 5 minutes break between the matches. After these 3 matches, we have a long pause e.g. 15 minuates where the ice gets cleaned before we start the next matches. For this scenario it would be good to have this module available.

**Reply by evroon** [reply]
Yes I agree a separate pause element would be useful, I was planning to implement that at some point as well.
